pyfar/classes/coordinates.py

```diff
@@ -2807,12 +2807,146 @@ def __eq__(self, other):
         return (eq_x & eq_y & eq_z).all() & eq_weights & eq_comment \
             & eq_sh_order & eq_system
 
+    def __add__(self, other):
+        """Add two Coordinates objects."""
+        return _arithmetics(self, other, 'add')
+
+    def __radd__(self, other):
+        """Add two Coordinates objects."""
+        return _arithmetics(other, self, 'add')
+
+    def __sub__(self, other):
+        """Subtract two Coordinates objects."""
+        return _arithmetics(self, other, 'sub')
+
+    def __rsub__(self, other):
+        """Subtract two Coordinates objects."""
+        return _arithmetics(other, self, 'sub')
+
+    def __mul__(self, other):
+        """Multiply two Coordinates objects."""
+        return _arithmetics(self, other, 'mul')
+
+    def __rmul__(self, other):
+        """Multiply two Coordinates objects."""
+        return _arithmetics(other, self, 'mul')
+
+    def __div__(self, other):
+        """Divide one Coordinates objects."""
+        return _arithmetics(self, other, 'div')
+
+    def __truediv__(self, other):
+        """Divide one Coordinates objects."""
+        return _arithmetics(self, other, 'div')
+
+    def __rtruediv__(self, other):
+        """Divide two Coordinates objects."""
+        return _arithmetics(other, self, 'div')
+
+    def __rdiv__(self, other):
+        """Divide two Coordinates objects."""
+        return _arithmetics(other, self, 'div')
+
+    def dot(self, other):
+        """Dot product two Coordinates objects."""
+
+        if not isinstance(other, Coordinates):
+            raise TypeError(
+                "Dot product is only possible with Coordinates objects.")
+
+        return self.x * other.x + self.y * other.y + self.z * other.z
```

Comment:
For discussion: should we encapsule all arithmetic operations and broadcasting in `try` statements to raise more informative errors?